conn_test.go

```diff
@@ -55,3 +55,28 @@ func TestRecurringReAuthHang(t *testing.T) {
 		t.Fatal(ctx.Err())
 	}
 }
+
+func TestDeadlockInClose(t *testing.T) {
+	c := &Conn{
+		shouldQuit:     make(chan struct{}),
+		connectTimeout: 1 * time.Second,
+		sendChan:       make(chan *request, sendChanSize),
```

Comment:
nit: since this test is aiming to test the condition of this channel being full, we can just have a size of 1. or a static size here.

the use of the variable is distracting to the conditions we are aiming to assert.